include/deal.II/base/utilities.h

```diff
@@ -429,12 +429,26 @@ namespace Utilities
 
   /**
    * Given an arbitrary object of type T, use boost::serialization utilities
-   * to pack the object into a vector of characters. The object can be unpacked
-   * using the Utilities::unpack function below.
+   * to pack the object into a vector of characters and append it to the
+   * given buffer. The object can be unpacked using the Utilities::unpack
+   * function below.
    *
    * If the library has been compiled with ZLIB enabled, then the output buffer
    * is compressed.
    *
+   * If many consecutive calls with the same buffer are considered, it is
+   * recommended for reasons of performance to ensure that its capacity is
+   * sufficient.
+   *
+   * @author Timo Heister, Wolfgang Bangerth, 2017.
+   */
+  template <typename T>
+  void pack (const T &object, std::vector<char> &dest_buffer);
```

Comment:
in current state the old `pack` which returns is not documented (you simply define at at the end of the this patch).
I would say you shall declare the old one here and document it, something like "Similar to the above but returns the buffer without appending".